action.yaml

```
@@ -0,0 +1,13 @@
+name: Setup go env using go version from go.mod
+description: Fetch go version from go.mod file, set it as an env var and use it to setup go env
```

No need to go into the implementation details in the description 
```suggestion
description: Fetch Go version from the go.mod file and use it to setup Go environment
```